lib/spack/spack/solver/asp.py

```diff
@@ -801,6 +831,14 @@ def on_model(model):
 
         timer.start("solve")
         solve_result = self.control.solve(**solve_kwargs)
+
+        if solve_result.satisfiable and self._model_has_cycles(models):
+            warnings.warn(f"cycles detected, falling back to slower algorithm [specs={specs}]")
```

Comment:
This isn't a warning -- it should just be a debug message b/c the user shouldn't have to do anything or worry about it. They'll see slower concretization time.

We *could* make it a regular message but I think "falling back to slower algorithm" is too much detail for that. Pretty convinced this belongs as a debug message.

Reply:
Done in 4512233c8accdc04ca04e9c77ff63788471acd40